src/coreclr/jit/hwintrinsic.h

```diff
@@ -191,6 +190,9 @@ enum HWIntrinsicFlag : unsigned int
     // The intrinsic uses a mask in arg1 to select elements present in the result, and must use a low register.
     HW_Flag_LowMaskedOperation = 0x40000,
 
+    // The intrinsic has an enum operand. Using this implies HW_Flag_HasImmediateOperand.
+    HW_Flag_HasEnumOperand = 0x80000,
```

Comment:
It's not clear to me why we need this to differentiate from `HasImmediateOperand`?

Reply:
> It's not clear to me why we need this in addition to `HasImmediateOperand`?

This is just for the hwintrinsiccodegen - so that it can pick the correct `emitIns_` function to call (namely `emitIns_R_PATTERN`).

Although, given an immediate and enum are effectively the same, I wonder if we should eventually do some refactoring so that it goes via `emitIns_R_I()` using an `int` and then remove `emitIns_R_PATTERN()`.